src/L07-24/TrainList.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace L07_24
+{
+    class TrainsList
+    {
+        private int trainsMaxNumber;
+        private Train[] trains;
+        private int trainsNumber = 0;
+        public TrainsList(int trainsMaxNumber, bool immediatelyFullComplete = false)
+        {
+            this.trainsMaxNumber = trainsMaxNumber;
+            trains = new Train[trainsMaxNumber];
+
+            if (immediatelyFullComplete)
+            {
+                for (int i = 0; i < this.trainsMaxNumber; i++)
+                {
+                    AddTrain();
+                }
+                SortByTrainNumber();
+            }
+        }
+        public void AddTrain()
+        {
+            if (trainsNumber >= trainsMaxNumber)
+            {
+                return;
+            }
+            trains[trainsNumber] = TrainFactory.CreateTrain(trainsNumber);
+            trainsNumber++;
+        }
+        class TrainNumberComparer : IComparer<Train>
+        {
+            public int Compare(Train train1, Train train2)
```

Comment:
try this https://learn.microsoft.com/ru-ru/dotnet/api/system.linq.enumerable.orderby?view=net-6.0